Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseCollectionStateManager.cs

```diff
@@ -180,6 +233,37 @@ public TimeSpan UpdateState(string instrumentationKey)
             return this.DetermineBackOffs();
         }
 
+        private void UpdateConfiguration(CollectionConfigurationInfo configurationInfo)
+        {
+            // we only get here if Etag in the header is different from the current one, but we still want to check if Etag in the body is also different
+            if (configurationInfo != null && !string.Equals(configurationInfo.ETag, this.currentConfigurationETag, StringComparison.Ordinal))
+            {
+                this.collectionConfigurationErrors.Clear();
```

Comment:
Oh, I think I misunderstood. Are you talking about this.collectionConfigurationErrors? It's just a list of errors that we need to update, it's not even the configuration. I have no reason to create a new instance of a List<> when I can just Clear it.